docs/api/admin/tags-api.md

```diff
@@ -0,0 +1,174 @@
+---
+id: tags
+title: /api/admin/tags
+---
+
+> In order to access the admin API endpoints you need to identify yourself. If you are using the `insecure` authentication method, you may use [basic authentication](https://en.wikipedia.org/wiki/Basic_access_authentication) to identify yourself.
+
+### Create a new tag
+
+`POST https://unleash.host.com/api/admin/tags`
+
+Creates a new tag without connecting it to any other object, can be helpful to build an autocomplete list.
+
+**Body**
+
+```json
+{
+  "value": "MyTag",
+  "type": "simple"
+}
+```
+
+### Notes
+
+- `type` must exist in tag-types
+
+### List tags
+
+`GET https://unleash.host.com/api/admin/tags`
+
+This endpoint is the one all admin UIs should use to fetch all available tags from the _unleash_server_. The response returns all tags.
+
+**Example response:**
+
+```json
+{
+  "version": 1,
+  "tags": [
+    {
+      "id": 5,
+      "value": "Team-Red",
+      "type": "simple"
+    },
+    {
+      "id": 9,
+      "value": "Team-Green",
+      "type": "simple"
+    },
+    {
+      "id": 19,
+      "value": "DecemberExperiment",
+      "type": "simple"
+    },
+    {
+      "id": 25,
+      "value": "#team-alert-channel",
+      "type": "slack"
+    }
+  ]
+}
+```
+
+### List tags by type
+
+`GET: https://unleash.host.com/api/admin/tags/:type`
+
+Lists all tags of `:type`. If none exist, returns the empty list
+
+**Example response to query for https://unleash.host.com/api/admin/tags/simple**
+
+```json
+{
+  "version": 1,
+  "tags": [
+    {
+      "id": 5,
+      "value": "Team-Red",
+      "type": "simple"
+    },
+    {
+      "id": 9,
+      "value": "Team-Green",
+      "type": "simple"
+    },
+    {
+      "id": 19,
+      "value": "DecemberExperiment",
+      "type": "simple"
+    }
+  ]
+}
+```
+
+### Fetching Tag types
+
+`GET: http://unleash.host.com/api/admin/tag-types`
+
+Used to fetch all types the server knows about. This endpoint is the one all admin ui should use to fetch all available tag types from the _unleash-server_. The response returns all tag types. Any server wil have _at least_ one configured tag type (the `simple` type). A tag type will have a `type`, `description`, `icon` map
+
+**Example response:**
+
+```json
+{
+  "version": 1,
+  "tagTypes": [
+    {
+      "name": "simple",
+      "description": "Arbitrary tags. Used to simplify filtering of features",
+      "icon": "#"
+    }
+  ]
+}
+```
+
+### Get a single tag type
+
+`GET: https://unleash.host.com/api/admin/tag-types/simple`
+
+Used to fetch details about a specific tag-type. This is mostly provded to make it easy to debug the API and should not be used by the client implementations.
+
+**Example response:**
+
+```json
+{
+  "version": 1,
+  "tagType": {
+    "name": "simple",
+    "description": "Some description",
+    "icon": "Some icon",
```

Comment:
what do we support in regards of icons? A valid [material-ui icon name](https://material.io/resources/icons/?style=baseline)? A url? An inline svg?

Reply:
After testing a bit in unleash-frontend with Chip and ChipContact from react-mdl, ChipContact does not display an <img> object, so I've fallen back to exposing either icon or an upcased first letter from name.

 Maybe we should try to look at allowing material-ui icon names? 
Allowing arbitrary URLs does necessitate a better validation of the input to make sure we don't open an attack vector.